src/Model.ts

```diff
@@ -458,7 +470,14 @@ export default class Model<S extends SyncFuncConstraint> extends EventEmitter<Re
         lastSyncParams: this.lastSyncParams,
       });
       this.removeStream();
+
       const { data, sequenceId } = await this.syncFunc(...(this.lastSyncParams || ([] as unknown as Parameters<S>)));
+      if (sequenceId === undefined) {
+        const err = Error('sync function response: sequenceId is undefined');
```

Comment:
nit-pick: I found this wording a little terse / hard to immediately understand. 
Do you think we could make it something that reads a bit more like a sentence and less like an error message 😅 maybe: 
> The sync function returned an undefined sequenceId

Reply:
Good idea